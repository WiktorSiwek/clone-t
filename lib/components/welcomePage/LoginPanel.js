import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { changePassword, changeLogin } from '../user/userActions';
import store from '../rootState/rootStore';

const ContainerRow = styled.div`
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: center;
    padding-top: 30px;
    padding-left: 140px;
    height: 10vh;
`;

const ContainerColumn = styled.div`
    display: flex;
    flex-direction: column;
`;

const LoginInput = styled.input`
    width: 200px;
    padding: 12px;
    box-sizing: border-box;
    margin-right: 20px;
`;

const ForgotenPasswordLink = styled.a`
    text-decoration: none;
    color: #aab8c2;
    font-size: 14px;
`;

const LogInButton = styled.button`
    height: 42px;
    color: #1da1f2;
    background-color: #ffffff;
    padding: 12px;
    font-weight: bold;
    font-size: 14px;
    border-radius: 30px;
    border-style: solid;
    border-color: #1da1f2;
    border-width: 1px;
    white-space: nowrap;
    cursor: pointer;
    outline: none;
    &:hover {
        background-color: #eaf5fd;
    }
`;

export function LoginPanel(props) {
   const { login, password } = props;

   const handleLoginChange = event => {
      store.dispatch(changeLogin(event));
   };

   const handlePasswordChange = event => {
      store.dispatch(changePassword(event));
   };

   const handleLogIn = () => {
      // eslint-disable-next-line no-console
      console.log(`Zalogowano ${login} - ${password}`);
   };
   return (
      <ContainerRow>
         <LoginInput type="text" placeholder="Phone, email, or username" value={login} onChange={handleLoginChange} />
         <ContainerColumn>
            <LoginInput type="password" placeholder="Password" value={password} onChange={handlePasswordChange} />
            <ForgotenPasswordLink href="/begin_password_reset">Forgot password?</ForgotenPasswordLink>
         </ContainerColumn>
         <LogInButton onClick={handleLogIn}>Log in</LogInButton>
      </ContainerRow>
   );
}

const mapStateToProps = (state) => {
   const { login, password } = state.userState;
   return {
      login,
      password
   };
};

export default withRouter(connect(
   mapStateToProps,
)(LoginPanel));

