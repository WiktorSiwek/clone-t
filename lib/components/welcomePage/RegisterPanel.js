import React from 'react';
import styled from 'styled-components';

const ContainerColumn = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 350px;
  margin: auto;
  padding-bottom: 15vh;
`;

const Icon = styled.i`
  color: #1da1f2;
  font-size: 45px;
  margin-bottom: 8px;
  `;

const Header = styled.h1`
  font-size: 27px;
  line-height: 32px;
  font-weight: bold;
  `;

const Text = styled.h2`
  margin-top: 40px;
  font-size: 16px;
  font-weight: 700;
`;

const Button = styled.button`
  width: 350px;
  border-radius: 100px;
  cursor: pointer;
  font-size: 14px;
  font-weight: bold;
  line-height: 20px;
  padding: 6px 16px;
  text-align: center;
  outline: none;
  box-shadow: none;
  border-style: solid;
  border-width: 1px;
  border-color: #1da1f2;
  margin-top: 16px;
  background-color: ${props => props.signup ? '#1b95e0' : '#ffffff'};
  color: ${props => props.signup ? '#ffffff' : '#1b95e0'};
  &:hover {
    background-color: ${props => props.signup ? '#006DBF' : '#eaf5fd'};
  }
`;

function RegisterPanel() {
   return (
      <ContainerColumn>
         <Icon className="Icon Icon--bird" />
         <Header>See what's happening in the world right now</Header>
         <Text>Join Twitter today.</Text>
         <Button signup='true'>Sign Up</Button>
         <Button>Log In</Button>
      </ContainerColumn>
   );
}

export default RegisterPanel;
