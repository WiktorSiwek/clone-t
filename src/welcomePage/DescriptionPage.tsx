import { faComment, faSearch, faUsers } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  color: white;
  line-height: 400%;
  margin: auto;
  width: 400px;
  padding: 30vh;
  display: flex;
  flex-direction: row;
`;

const Column = styled.div`
  display: flex;
  flex-direction: column;
`;

const Text = styled.span`
  font-size: 17px;
  font-weight: bold;
  padding-left: 15px;
`;

const Icon = styled.span`
  font-size: 28px;
`;


export default function DescriptionPage() {
  return (
    <Container>
      <Column>
        <Icon>
          <FontAwesomeIcon icon={faSearch} />
        </Icon>
        <Icon>
          <FontAwesomeIcon icon={faUsers} />
        </Icon>
        <Icon>
          <FontAwesomeIcon icon={faComment} />
        </Icon>
      </Column>
      <Column>
        <Text>Obserwuj to, co Cię interesuje</Text>
        <Text>Dowiedz się, o czym rozmawiają inni.</Text>
        <Text>Dołącz do rozmowy.</Text>
      </Column>
    </Container>
  )
}

