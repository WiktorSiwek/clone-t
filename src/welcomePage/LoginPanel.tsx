import React from 'react';
import styled from 'styled-components';

const ContainerRow = styled.div`
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: center;
    padding-top: 30px;
`;

const ContainerColumn = styled.div`
    display: flex;
    flex-direction: column;
`;

const LoginInput = styled.input`
    width: 200px;
    padding: 12px;
    box-sizing: border-box;
    margin-right: 20px;
`;

const ForgotenPasswordLink = styled.a`
    text-decoration: none;
    color: #aab8c2;
    font-size: 14px;
`;

const LogInButton = styled.button`
    height: 42px;
    color: #1da1f2;
    background-color: #ffffff;
    padding: 12px;
    font-weight: bold;
    font-size: 16px;
    border-radius: 30px;
    border-style: solid;
    border-color: #1da1f2;
    border-width: 1px;
    cursor: pointer;
    outline: none;
    &:hover {
        background-color: #eaf5fd;
    }
`;


export default class LoginPanel extends React.Component {
    public render() {
        return (
            <ContainerRow>
                    <LoginInput type="text" placeholder="Phone, email, or username" />
                    <ContainerColumn>
                        <LoginInput type="password" placeholder="Password" />
                        <ForgotenPasswordLink href="/begin_password_reset">Forgot password?</ForgotenPasswordLink>
                    </ContainerColumn>
                    <LogInButton>Log in</LogInButton>
            </ContainerRow>
        )
    }
}
