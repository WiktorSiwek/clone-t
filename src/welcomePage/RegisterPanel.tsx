import { faDove } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import styled from 'styled-components';

const ContainerColumn = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 350px;
  margin: auto;
  margin-bottom: 45vh;
`;

const Icon = styled.i`
  color: #1da1f2;
  font-size: 45px;
  margin-bottom: 8px;
  `;

const Header = styled.h1`
  font-size: 27px;
  line-height: 32px;
  font-weight: bold;
  `;

const Text = styled.h2`
  margin-top: 40px;
  font-size: 16px;
  font-weight: 700;
`;

const Button = styled.button`
  width: 350px;
  border-radius: 100px;
  cursor: pointer;
  font-size: 14px;
  font-weight: bold;
  line-height: 20px;
  padding: 6px 16px;
  text-align: center;
  outline: none;
  box-shadow: none;
  border-style: solid;
  border-width: 1px;
  border-color: #1da1f2;
  margin-top: 16px;
`;

const SingUpButton = styled(Button)`
  color: #ffffff;
  background-color: #1B95E0;
  &:hover {
    background-color: #006DBF;
  }
  `;

const LogInButton = styled(Button)`
  color: #1b95e0;
  background-color: #ffffff;
  &:hover {
    background-color: #eaf5fd;
  }
`;

export default function RegisterPanel() {
  return (
    <ContainerColumn>
      <Icon className="Icon Icon--bird" />
      <Header>See what's happening in the world right now</Header>
      <Text>Join Twitter today.</Text>
      <SingUpButton>Sign Up</SingUpButton>
      <LogInButton>Log In</LogInButton>
    </ContainerColumn>
  )
}
