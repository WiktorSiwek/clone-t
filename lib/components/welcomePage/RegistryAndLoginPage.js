import React from 'react';
import styled from 'styled-components';
import LoginPanel from './LoginPanel';
import RegisterPanel from './RegisterPanel';

const Container = styled.div`
    background-color: #ffffff;
    /* height: 100vh; */
    height:100%;
`;

const Center = styled.div`
   display: flex;
   align-items: center;
   width: 100%;
   height: calc(90vh - 30px);
`;

function RegistryAndLoginPage() {
   return (
      <Container>
         <LoginPanel />
         <Center>
            <RegisterPanel />
         </Center>
      </Container>
   );
}

export default RegistryAndLoginPage;
