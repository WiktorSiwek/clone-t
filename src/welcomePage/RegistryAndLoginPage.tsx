import React from 'react'
import styled from 'styled-components';
import LoginPanel from './LoginPanel';
import RegisterPanel from './RegisterPanel';

const Container = styled.div`
    background-color: #ffffff;
    height: 100%;
`;

const Center = styled.div`
    display: flex;
    align-items: center;
    width: 100%;
    height: 100%;
`;

function RegistryAndLoginPage() {
  return (
    <Container>
      <LoginPanel />
      <Center>
        <RegisterPanel />
      </Center>
    </Container>
  )
}

export default RegistryAndLoginPage;
