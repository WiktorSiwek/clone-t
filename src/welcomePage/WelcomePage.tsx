import React, { Component } from 'react'
import styled from 'styled-components';

import DescriptionPage from './DescriptionPage';

const HalfScreen = styled.div`
    width: 50vw;
    height: 100vh;
`;

export default class WelcomePage extends Component {
    public render() {
        return (
        <HalfScreen>
            <DescriptionPage />
        </HalfScreen>
        )
    }
}
