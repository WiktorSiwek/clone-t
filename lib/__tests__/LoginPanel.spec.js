import React from 'react';
import { LoginPanel } from '../components/welcomePage/LoginPanel';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

describe('Login Panel', () => {
   let wrapper;
   let middlewares;
   let mockStore;
   
   beforeEach(() => {
      // wrapper = shallow(<LoginPanel login="admin" password="password" />); //w enzyme 3 juz nie ma snpashotow
      wrapper = renderer.create(<LoginPanel login="admin" password="password" />);
      middlewares = [];
      mockStore = configureStore(middlewares);
   });

   it('should render as expected', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
   });

   // describe('handleLoginChange', () => {
   //    it('should dispatch proper action', () => {
   //       console.log(wrapper.find('button'));
   //       expect(wrapper.find('button')).to.have.lengthOf(2);
   //    });
   // });
   describe('handleLoginChange', () => {
      it('should dispatch proper action', () => {
         const initialState = {};
         const store = mockStore(initialState);
         const event = jest.fn();
         const changePassword = () => ({type: 'CHANGE_PASSWORD', payload: event});


         expect(wrapper.find('input')).toHaveLength(2);
         // store.dispatch(changePassword(event));

         // const actions = store.getActions();
         // const expectedPayload = {type: 'CHANGE_PASSWORD', payload: event};
         // expect(actions[0]).toEqual(expectedPayload);
      });
   });

});

// Znajdz input, zapelnij go, sprawdz czy zdispatchowalo odpowiednia akcje

///  Zastapic enzyme react-testing-library

