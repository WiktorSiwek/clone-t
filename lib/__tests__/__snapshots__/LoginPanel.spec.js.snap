// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Panel should render as expected 1`] = `
<div
  className="sc-bdVaJa dJRLBJ"
>
  <input
    className="sc-htpNat cvMTL"
    onChange={[Function]}
    placeholder="Phone, email, or username"
    type="text"
    value="admin"
  />
  <div
    className="sc-bwzfXH bRskXu"
  >
    <input
      className="sc-htpNat cvMTL"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="password"
    />
    <a
      className="sc-bxivhb fjBXtV"
      href="/begin_password_reset"
    >
      Forgot password?
    </a>
  </div>
  <button
    className="sc-ifAKCX bwavoa"
    onClick={[Function]}
  >
    Log in
  </button>
</div>
`;
